lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -106,15 +106,21 @@ export class PublicClientApplication {
      * containing data from the server (returned with a null or non-blocking error).
      */
     async handleRedirectCallback(authCallback: AuthCallback): Promise<void> {
-        if(!authCallback) {
+        if (!authCallback) {
             throw BrowserConfigurationAuthError.createInvalidCallbackObjectError(authCallback);
         }
 
         this.authCallback = authCallback;
         const { location: { hash } } = window;
-        if (UrlString.hashContainsKnownProperties(hash)) {
-            const interactionHandler = new RedirectHandler(this.authModule, this.browserStorage, this.authCallback);
-            interactionHandler.handleCodeResponse(hash);
+        const cachedHash = this.browserStorage.getItem(TemporaryCacheKeys.URL_HASH);
+        try {
+            const interactionHandler = new RedirectHandler(this.authModule, this.browserStorage, this.authCallback, this.config.auth.navigateToLoginRequestUrl);
+            const responseHash = UrlString.hashContainsKnownProperties(hash) ? hash : cachedHash;
+            if (responseHash) {
+                this.authCallback(null, await interactionHandler.handleCodeResponse(responseHash));
```

Comment:
Functionally this is fine, kinda weird though to put the `await` inline, makes it a little harder to understand. Not a big deal though.

Reply:
happy to change it, it just seemed a little more concise.